modeling/inference_models/4bit_hf_torch/class.py

```diff
@@ -0,0 +1,227 @@
+from __future__ import annotations
+
+import os
+import glob
+import json
+import torch
+import re
+import shutil
+import sys
+from typing import Union
+
+from transformers import GPTNeoForCausalLM, AutoTokenizer, LlamaTokenizer
+from hf_bleeding_edge import AutoModelForCausalLM
+
+import utils
+import modeling.lazy_loader as lazy_loader
+import koboldai_settings
+from logger import logger, set_logger_verbosity
+
+try:
+    import breakmodel
+except ModuleNotFoundError as e:
+    # Breakmodel is only expected to work on GPU
+    if not utils.koboldai_vars.use_colab_tpu:
+        raise e
+
+from modeling.inference_models.hf_torch import HFTorchInferenceModel
+from modeling.tokenizer import GenericTokenizer
+
+# 4-bit dependencies
+import gptq
+from pathlib import Path
+from gptq.gptj import load_quant as gptj_load_quant
+from gptq.gptneox import load_quant as gptneox_load_quant
+from gptq.llama import load_quant as llama_load_quant
+from gptq.opt import load_quant as opt_load_quant
+from gptq.mpt import load_quant as mpt_load_quant
+from gptq.offload import load_quant_offload
+
+
+model_backend_name = "Huggingface GPTQ"
+
+
+def load_model_gptq_settings(path):
+    try:
+        js = json.load(open(path + "/config.json", "r"))
+    except Exception as e:
+        return False, -1, -1, False, -1
+
+    gptq_model = False
+    gptq_bits = -1
+    gptq_groupsize = -1
+    gptq_file = False
+    gptq_version = -1
+
+    gptq_legacy_files = glob.glob(os.path.join(path, "4bit*.pt")) + glob.glob(os.path.join(path, "4bit*.safetensors"))
```

Comment:
It doesn't, because that is the legacy format and was supposed to be replaced with parameters in the config.json. But you're right that it would be helpful to recognize more stuff from filenames.